Marketplace/controllers/api/apiProductController.js

```diff
@@ -1,11 +1,77 @@
-const db = require("../../database/models");
-const entidad = db.Producto;
+const productoRepository = require("../../repositories/productoRepository");
+const categoriaRepository = require("../../repositories/categoriaRepository");
 
 module.exports = {
-	listado: (req, res) => {
-
+	listado: async (req, res) => {
+		let listado = null;
+		listado = await productoRepository.ObtenerTodas();
+		// *** PRODUCTOS ***
+		let productos = [];
+		listado.map(n => {
+			let aux=[]
+			n.imagenes.map(m => aux.push(m.ruta));
+			productos.push({
+				id: n.id,
+				nombre: n.nombre,
+				descripcion: n.descripcion,
+				categoria: n.categoria.nombre,
+				imagenes: aux,
+				url: "/api/productos/" + n.id,
+			});
+		});
+		let resumenProductos = {
+			cantidad: listado.length,
+			detalle: productos,
+		};
+		// *** CATEGORÍAS ***
+		listado = await categoriaRepository.ObtenerTodas();
+		let categorias = [];
+		listado.map(n => {
+			let aux = [];
+			n.productos.map(m => {
+				aux.push({
+					id: m.id,
+					nombre: m.nombre,
+					url: "/api/productos/" + m.id,
+				});
+			});
+			categorias.push({
+				id: n.id,
+				nombre: n.nombre,
+				cantidad_de_productos: n.productos.length,
+				productos: aux,
+			});
+		});
+		let resumenCategorias = {
+			cantidad: listado.length,
+			detalle: categorias,
+		};
+		// *** FINAL ***
+		let respuesta = {
+			productos: resumenProductos,
+			categorias: resumenCategorias,
+		};
+		res.json(respuesta);
 	},
-	detalle: (req, res) => {
 
+	detalle: async (req, res) => {
+		let listado = null;
+		listado = await productoRepository.ObtenerPorId(req.params.id);
+		// *** Proceso de la info ***
+		// Obtener el nombre y URL de las imagenes
+		let aux = []
+		listado.imagenes.map(n => aux.push(n.ruta))
+		let producto = {
+			id: listado.id,
+			nombre: listado.nombre,
+			descripcion: listado.descripcion,
+			categoria: listado.categoria.nombre,
+			marca: listado.marcas.nombre,
+			modelo: listado.modelos.nombre,
+			precio: listado.precio,
+			stock: listado.stock_disponible,
+			imagenes: aux,
+		};
```

Comment:
La variable auxiliar para mi esta de mas, seria mas sensillo hacer el amp directo en la creacion de objeto, algo asi:
```suggestion
		let producto = {
			id: listado.id,
			nombre: listado.nombre,
			descripcion: listado.descripcion,
			categoria: listado.categoria.nombre,
			marca: listado.marcas.nombre,
			modelo: listado.modelos.nombre,
			precio: listado.precio,
			stock: listado.stock_disponible,
			imagenes: listado.imagenes.map(x => x.ruta),
		};
```

Reply:
Muy de acuerdo. No se me había ocurrido hacerlo de esa manera.  Lo vas a ver así cuando actualice el pull.